numpy-dev using a dtype with a subarray field is deprecated

<!-- This comments are hidden when you submit the issue,
so you do not need to remove them! -->

<!-- Please be sure to check out our contributing guidelines,
https://github.com/astropy/astropy/blob/master/CONTRIBUTING.md .
Please be sure to check out our code of conduct,
https://github.com/astropy/astropy/blob/master/CODE_OF_CONDUCT.md . -->

<!-- Please have a search on our GitHub repository to see if a similar
issue has already been posted.
If a similar issue is closed, have a quick look to see if you are satisfied
by the resolution.
If not please go ahead and open an issue! -->

<!-- Please check that the development version still produces the same bug.
You can install development version with
pip install git+https://github.com/astropy/astropy
command. -->

- [x] Remember to undo #10863 ! See #10966

### Description
<!-- Provide a general description of the bug. -->

@astrofrog reported `DeprecationWarning: using a dtype with a subarray field is deprecated.` warnings using numpy-dev in v4.0.x ([example log](https://travis-ci.org/github/astropy/astropy/jobs/734002575)). Since v4.0.x is installing numpy from master, while master installs from "nightly" (i.e., weekly) wheel, we won't see it until numpy's wheel is updated.

Caused by numpy/numpy#17419

### Steps to Reproduce
<!-- Ideally a code example could be provided so we can run it ourselves. -->
<!-- If you are pasting code, use triple backticks (```) around
your code snippet. -->
<!-- If necessary, sanitize your screen output to be pasted so you do not
reveal secrets like tokens and passwords. -->

I think this might trigger the warning:

```python
>>> import numpy as np
>>> from astropy.io.votable import converters, tree
>>> config = {'verify': 'exception'}
>>> field = tree.Field(None, name='c', datatype='floatComplex', arraysize='2x3*', config=config)
>>> c = converters.get_converter(field, config=config)
>>> c._base.format
'(2,)c8'
>>> default = np.array([], dtype=c._base.format)  # Warning?
```
```
E       DeprecationWarning: using a dtype with a subarray field is deprecated.
This can lead to inconsistent behaviour due to the resulting dtype being different from the input dtype.
You may try to use `dtype=dtype.base`, which should give the same result for most inputs,
but does not guarantee the output dimensions to match the subarray ones.  (Deprecated NumPy 1.20)
```

In this case, `c._base.format` is a string, so I am not sure what `dtype.base` means in this context. @mhvk , any advise?

--- a/setup.cfg
+++ b/setup.cfg
@@ -131,7 +131,6 @@ filterwarnings =
     ignore:Unknown pytest.mark.mpl_image_compare:pytest.PytestUnknownMarkWarning
     ignore:Unknown config option:pytest.PytestConfigWarning
     ignore:matplotlibrc text.usetex:UserWarning:matplotlib
-    ignore:using a dtype with a subarray field is deprecated:DeprecationWarning
     # Triggered by ProgressBar > ipykernel.iostream
     ignore:the imp module is deprecated:DeprecationWarning
     # toolz internal deprecation warning https://github.com/pytoolz/toolz/issues/500

